nntrainer/utils/base_properties.h

```diff
@@ -260,6 +260,23 @@ template <typename T> class Property {
   std::unique_ptr<T> value; /**< underlying data */
 };
 
+/**
+ * @brief base template to store pack of enumeration information specialization
+ * @details below need to be implemented for enum_info
+ * enum class Enum
+ * static constexpr std::initializer_list<Enum> EnumList,
+ * static constexpr const char *EnumStr[]
+ */
+template <typename T> struct enum_info {};
+
+/**
+ * @brief enum property
+ *
+ * @tparam T underlying type to query enum_info
+ */
+template <typename T>
+class EnumProperty : public Property<typename enum_info<T>::Enum> {};
```

Comment:
This should be changed just EnumInfo types, and injected to the properties to make it reusable.